eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentPacket.scala

```diff
@@ -249,4 +250,15 @@ object OutgoingPacket {
     CMD_ADD_HTLC(replyTo, firstAmount, paymentHash, firstExpiry, onion.packet, Origin.Hot(replyTo, upstream), commit = true) -> onion.sharedSecrets
   }
 
+  def buildHtlcFailure(nodeSecret: PrivateKey, cmd: CMD_FAIL_HTLC, add: UpdateAddHtlc): Try[UpdateFailHtlc] = {
```

Comment:
turns out this method gets inlined and is not visible to plugin, `@noinline` does not help. Are there any other ways to prevent inlining? Otherwise it makes no sense to have it and code can be copy-pasted in plugin.

Reply:
That sounds very surprising...why would this one be inlined and not all others? Are you sure you're referencing the right jar?

Reply:
yes, I'm sure since `AbstractCommitments` are visible. I've also clicked "decompile to java" in `Commitments.scala` and the method body, not a function call can be seen in `sendFail`.

Reply:
Damn this is weird...and are you able to see the other functions in the `OutgoingPacket` object?

Reply:
The only explanation I see is that it's used in only one place, maybe if you add a unit test for it it won't be inlined?

Reply:
Adding a method to tests solved an issue.